client/js/widgets/seat.js

```diff
@@ -33,8 +33,13 @@ class Seat extends Widget {
       displayedText = displayedText.replaceAll('playerName',this.get('player'))
       setText(this.domElement, displayedText);
     }
-    if(delta.player !== undefined)
-      this.updateLinkedWidgets();
+    if(delta.player !== undefined || delta.score !== undefined) {
+      const scoreboard = widgetFilter(w => w.get('type') == 'scoreboard');
+      if(scoreboard.length > 0)
+        scoreboard[0].applyDeltaToDOM(delta);
```

Comment:
Could loop through all the scoreboard widgets calling `applyDeltaToDOM()` rather than only supporting one scoreboard.

I'm not sure that we want to pass the seat's `delta` to the scoreboard widgets `applyDeltaToDOM()`; scoreboard's call to `super.applyDeltaToDOM(delta)` might do some unexpected things with the values. Should probably call with an empty object.

Reply:
This has all changed more or less as suggested.